lib/Doctrine/DBAL/Sharding/ShardChoser/ShardChoser.php

```diff
@@ -12,10 +12,6 @@
 {
     /**
      * Picks a shard for the given distribution value.
-     *
-     * @param string|int $distributionValue
-     *
-     * @return string|int
      */
-    public function pickShard($distributionValue, PoolingShardConnection $conn);
+    public function pickShard(string $distributionValue, PoolingShardConnection $conn) : string;
```

Comment:
Dropping `int` is intentional BC break?

Reply:
No, it should remain `string|int`.

Reply:
Well, if `$distributionValue` is the partition key what determines the shard, it can be anything including a list of different values. In light of this, the implementation of `MultiTenantShardChoser` looks questionable. Should we just declare it `mixed`?